awscrt/auth.py

```diff
@@ -255,6 +255,30 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
         return cls(binding)
 
+    @classmethod
+    def new_py_provider(cls, py_provider):
```

Comment:
consider adding an optional (default = true) argument to decorate with a caching provider

Reply:
Still not fully understand the cache provider... Can the caching provider be added as another provider in the chain provider list?

Reply:
Yes.

Reply:
So, why we need to add an option here?🤔

Reply:
I guess you don't need to but then the user needs to worry about all the caching themslves.  They have no way of building a provider chain nor a caching provider, so it will all have to come from python.

Reply:
me and Bret chatted.
We'll add the cached_provider in a separate PR